### PR TITLE
Yaml layer: enable flycheck

### PR DESCRIPTION
**Pull request by rski**

### CODE REVIEW

File: layers/+lang/yaml/packages.el [hunk 1]
```diff
@@ -9,14 +9,19 @@
...
 (defun yaml/post-init-company ()
   (spacemacs|add-company-hook yaml-mode))
 
+(defun yaml/post-init-flycheck ()
+  (spacemacs/add-flycheck-hook 'yaml-mode))
+
 (defun yaml/init-yaml-mode ()
   "Initialize YAML mode"
   (use-package yaml-mode
+    :defer t
```
Comment (TheBB) [inline layers/+lang/yaml/packages.el]: `:mode` implies `:defer`, so no need for this.

Comment (rski) [inline layers/+lang/yaml/packages.el]: oh, i see, one sec then

**Comment by robbyoconnor** [comment]
@TheBB -- actually all flycheck checkers require external tools. Literally all flycheck does is parse the output from the various tools...

File: layers/+lang/yaml/config.el [hunk 2]
```diff
@@ -1,4 +1,4 @@
-;;; config.el --- YAML Layer packages File for Spacemacs
+;;; config.el --- YAML Layer configuration File for Spacemacs
```
Comment (d12frosted) [inline layers/+lang/yaml/config.el]: Thanks for fixing this line. But it should be `config` 😸

Comment (rski) [inline layers/+lang/yaml/config.el]: I can fix it, but I just looked it up from other layers:
```
~/Code/spacemacs/spacemacs on yaml_flycheck ⌚ 20:34:53
$ ag "config file"| wc -l
11

~/Code/spacemacs/spacemacs on  yaml_flycheck ⌚ 20:35:00
$ ag "configuration file"| wc -l
67
```

Comment (d12frosted) [inline layers/+lang/yaml/config.el]: Hm... indeed 😸

**Comment by d12frosted** [comment]
Thank you :+1: Cherry-picked into `develop`, you can safely delete your branch.

**Pull request merged**